nav2_costmap_2d/include/nav2_costmap_2d/costmap_filters/filter_values.hpp

```diff
@@ -0,0 +1,62 @@
+/*********************************************************************
+ *
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2020 Samsung Research Russia
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of the <ORGANIZATION> nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Alexey Merzlyakov
+ *********************************************************************/
+
+#ifndef NAV2_COSTMAP_2D__FILTER_VALUES_HPP_
+#define NAV2_COSTMAP_2D__FILTER_VALUES_HPP_
+
+/** Provides constants used in costmap filters */
+
+namespace nav2_costmap_2d
+{
+
+/** Types of costmap filter */
+static constexpr uint8_t KEEPOUT_FILTER = 0;
+static constexpr uint8_t SPEED_FILTER_PERCENT = 1;
+static constexpr uint8_t SPEED_FILTER_ABSOLUTE = 2;
+
+/** Default values for base and multiplier */
+static constexpr double BASE_DEFAULT = 0.0;
+static constexpr double MULTIPLIER_DEFAULT = 1.0;
+
+/** Speed filter constants */
+static constexpr int8_t SPEED_MASK_UNKNOWN = -1;
```

Comment:
From other comments, I think the UNKNOWN should be removed, I don't think that's sensible for the class of data that speed zones are. Occupancy grids track unknown space for exploration purposes.

Reply:
Okay, I will remove it from filter's logic. However, this constant still might be useful just for the case. Let's imagine, that developer gave the mask with unknown areas on it. If we will not have this constant, the Speed Filter will produce "-1%" of speed restriction in that case. So, I believe the best way - is to remain `SPEED_MASK_UNKNOWN` constant in header, but do nothing in speed_filter body if we meet this on map.

Reply:
Now, we are do nothing, if filter mask is unknown: https://github.com/ros-planning/navigation2/commit/34e694dbee7af9f805d2a7038b292466e44dc104#diff-d344e376ecbc8e5ecd5045731e13855c6d2a9a955cdaee9b31dddbbee7681466R271

Reply:
See above comment, same with the odd branching issue

Reply:
Ok, I think an error log here would be good to show people that unknown is an invalid value.

Reply:
Added error message in https://github.com/ros-planning/navigation2/pull/2074/commits/c4584cd1226c18ac363d69b89d0e91737ab05063#diff-d344e376ecbc8e5ecd5045731e13855c6d2a9a955cdaee9b31dddbbee7681466R274